contracts/babylon/src/contract.rs

```diff
@@ -269,6 +271,44 @@ pub fn execute(
             // TODO: Add events
             Ok(res)
         }
+        ExecuteMsg::SendRewards { fp_distribution } => {
+            let cfg = CONFIG.load(deps.storage)?;
+            // Assert the funds are there
+            must_pay(&info, &cfg.denom)?;
+            // Assert the sender is right
+            let btc_finality = cfg
+                .btc_finality
+                .ok_or(ContractError::BtcFinalityNotSet {})?;
+            if info.sender != btc_finality {
+                return Err(ContractError::Unauthorized {});
+            }
+            // Build the memo payload
+            let memo_msg = to_json_string(&fp_distribution)?;
+            // Route to babylon over IBC
+            let channel = IBC_CHANNEL.load(deps.storage)?;
```

Comment:
Hmm I guess `IBC_CHANNEL` is the channel that this contract involves, but not necessarily a channel for IBC transfer. Probably we need to confirm this in some e2e test.

Maybe as a part of the integration, one needs to establish an IBC channel for interchain transfer, and the contract side needs to remember this channel ID in the config, for facilitating the reward mechanism.

Reply:
Yes, this was my suspicion as well. Did it way for simplicity. Perhaps we can configure a single channel to do / support both, custom messages and ICS20 transfers?

Will write an e2e test and confirm this in any case.

Reply:
Finally was able to e2e test this.

Using a single channel doesn't work, as CosmWasm's `IbcMsg::Transfer` requires an established channel against the `ibctransfer` module, not the contract. :-/ I think this is a design decision that makes sense, as you don't want to establish a channel for every contract that needs / wants to use ICS20 transfers.

Reply:
> Using a single channel doesn't work, as CosmWasm's IbcMsg::Transfer requires an established channel against the ibctransfer module, not the contract address under wasm.

Yes that's right, even though we can wire up a IBC transfer packet in the contract side, babylon side's zone concerige won't recognise it.

If we really, really want to let zone concerige handle IBC tansfer, then zone concierge has to invoke IBC transfer module (in particular [this](https://github.com/cosmos/ibc-go/blob/91169d134d659b87acaf4903652cdcb1a55f5657/modules/apps/transfer/ibc_module.go#L193-L204) code snippet) to handle the IBC transfer packets. But I'm not sure if it's worth it

So I'd recommend we consider consolidating IBC channels as future work, and for now start with having 2 IBC channels. This also creates a clean separation between channels' functionalities following IBC channels' best practices